srv/_modules/sspl.py

```diff
@@ -44,17 +44,27 @@ def merge_health_map_schema(source_json="/tmp/resource_health_view.json"):
   health_map_file = healthmap_schema_pillar['filename']
 
   data = local.cmd('*', 'file.read', [source_json])
-  node1_data = json.loads(data["srvnode-1"])
-  node2_data = json.loads(data["srvnode-2"])
+  node_data = []
+  for node in __pillar__['cluster'].keys():
+    if "srvnode-" in node:
+      node_data = json.loads(data[node])
+
+  node_rack_nodes = None
+  for i, node_data in enumerate(node_rack_nodes):
+    if i == 0:
+      node_rack_nodes = (
+        node_data[i]["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]
+      )
+    else:
+      node_rack_nodes.update(
+        node_data[i]["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]
+      )
```

Comment:
In this context it is safe to use `node_data[0]["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]` as initial directory for `node_rack_nodes`. But such approach modifies in parallel the original object to which refers variable `node_data`

Reply:
@dmitrydb, I am not sure I understand, how original object is modified here.
Could you elaborate?

Reply:
It's because this assignment
```python
    if i == 0:
      node_rack_nodes = (
        node_data[i]["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]
      )
```
copies reference to the dict rather than the data of this dict.
It is better to illustrate this by example:

```python
>>> a = [{"nodes": {"node_1": 1, "node_2": 2} }, {"nodes": {"nodes_3": 3, "nodes_4": 4}}]
>>> for i, _ in enumerate(a):
...     if i == 0:
...             nodes_rack = a[i]["nodes"]
...     else:
...             nodes_rack.update(a[i]["nodes"])
... 
>>> a
[{'nodes': {'node_1': 1, 'node_2': 2, 'nodes_3': 3, 'nodes_4': 4}}, {'nodes': {'nodes_3': 3, 'nodes_4': 4}}]
>>> nodes_rack
{'node_1': 1, 'node_2': 2, 'nodes_3': 3, 'nodes_4': 4}
>>> id(nodes_rack)
139647028594752
>>> id(a[0]["nodes"])
139647028594752
>>> id(a[0]["nodes"]) == id(nodes_rack)
True
```